pkg/kube/controllers/boshdeployment/reconciler.go

```diff
@@ -135,24 +475,299 @@ func (r *ReconcileBOSHDeployment) Reconcile(request reconcile.Request) (reconcil
 		existingESts := &estsv1.ExtendedStatefulSet{}
-		err = r.client.Get(ctx, types.NamespacedName{Name: eSts.Name, Namespace: eSts.Namespace}, existingESts)
+		err := r.client.Get(ctx, types.NamespacedName{Name: eSts.Name, Namespace: eSts.Namespace}, existingESts)
 		if err != nil && apierrors.IsNotFound(err) {
-			r.log.Infof("Creating a new ExtendedStatefulset %s/%s for Deployment Manifest %s\n", eSts.Namespace, eSts.Name, instance.Name)
+			r.log.Infof("Creating a new ExtendedStatefulSet %s/%s for Deployment Manifest %s\n", eSts.Namespace, eSts.Name, instance.Name)
 
 			// Create the extended statefulset
 			err := r.client.Create(ctx, &eSts)
 			if err != nil {
 				r.recorder.Event(instance, corev1.EventTypeWarning, "CreateExtendedStatefulSetForDeployment Error", err.Error())
-				r.log.Errorf("Error creating ExtendedStatefulSet %s for deployment manifest %s: %s", eSts.Name, request.NamespacedName, err)
-				return reconcile.Result{}, errors.Wrap(err, "couldn't create an ExtendedStatefulSet for a BOSH Deployment")
+				r.log.Errorf("Error creating ExtendedStatefulSet %s for deployment manifest %s: %s", eSts.Name, instance.GetName(), err)
+				return errors.Wrap(err, "couldn't create an ExtendedStatefulSet for a BOSH Deployment")
 			}
 		}
 	}
 
-	return reconcile.Result{}, nil
+	instance.Status.State = deployingState
+
+	return nil
+}
+
+// actionOnDeploying check out deployment status
+func (r *ReconcileBOSHDeployment) actionOnDeploying(ctx context.Context, instance *bdc.BOSHDeployment, kubeConfigs *bdm.KubeConfig) error {
+	// TODO Check deployment
+	instance.Status.State = deployedState
+
+	return nil
+}
+
+// newExtendedJobTemplateForVariableInterpolation returns a job to interpolate variables
+func (r *ReconcileBOSHDeployment) newExtendedJobTemplateForVariableInterpolation(ctx context.Context, manifest *corev1.Secret, variables []esv1.ExtendedSecret, jobLabels map[string]string, secretLabels map[string]string, namespace string) (*ejv1.ExtendedJob, error) {
+	cmd := []string{"/bin/sh"}
+	args := []string{"-c", "cf-operator variable-interpolation --manifest /var/run/secrets/manifest/manifest.yaml --variables-dir /var/run/secrets/variables | base64 | tr -d '\n' | echo \"{\\\"interpolated-manifest.yaml\\\":\\\"$(</dev/stdin)\\\"}\""}
+
+	volumes := []corev1.Volume{
+		{
+			Name: string(manifest.GetUID()),
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: manifest.GetName(),
+					Items: []corev1.KeyToPath{
+						{
+							Key:  "manifest.yaml",
+							Path: "manifest.yaml",
+						},
+					},
+				},
+			},
+		},
+	}
+	volumeMounts := []corev1.VolumeMount{
+		{
+			Name:      string(manifest.GetUID()),
+			MountPath: "/var/run/secrets/manifest",
+			ReadOnly:  true,
+		},
+	}
+
+	for _, variable := range variables {
+		vars := Variables{}
+		varName := variable.GetLabels()["variableName"]
+
+		secretItems := []corev1.KeyToPath{}
+
+		// Check if variable already exists
```

Comment:
Don't check.
Just create the ejob - it will **eventually** run, once everything is generated.

Reply:
I think I should explain what func `newExtendedJobTemplateForVariableInterpolation` does:
For variable interpolation job, this job should get variable values as this structure: https://bosh.io/docs/variable-types

For example, this interpolated ssh var can be used as `diego_ssh_proxy_host_key.private_key`:
**structure 1**
```
diego_ssh_proxy_host_key:
  private_key: |
    *****
  public_key: |
    *****
  public_key_fingerprint: *****
```

And our extendSecret just generates a plain structure:
**structure 2**
```
data:
  SSHPrivateKey: ***
  SSHPublicKey: ***
  SSHFingerprint: ***
```
And that's why this func constructs a `***-gen` key to each variable and move **structure 2** to **structure 1**. From my point of view, replacing existing variable secrets just works. But we can discuss a better solution.

And for this reason, this func should wait for secret to be created and replace it.

Reply:
Ok, so this is just about how the data looks.
We do not want to process data here. 
We have 2 options:

1. Change the way `ExtendedSecrets` place data in the secrets
2. Mount the data as is, and move data processing to the actual interpolation command

Because our API is not stable, I'd like us to do option 1.
Another reason is that I don't like the usage of Capital Leters in those keys (they can become filenames when mounting the secret).